com.github.alexkdeveloper.web-app-creator.json

```diff
@@ -0,0 +1,41 @@
+{
+  "app-id": "com.github.alexkdeveloper.web-app-creator",
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "42",
+  "sdk": "org.gnome.Sdk",
+  "command": "com.github.alexkdeveloper.web-app-creator",
+  "finish-args": [
+     "--device=dri",
+     "--share=ipc",
+     "--socket=fallback-x11",
+     "--socket=wayland",
+     "--filesystem=home:ro",
+     "--filesystem=~/.local/share/applications"
```

Comment:
is there a reason for this?

Reply:
In the specified folder, the program saves the created icons for applications.

Reply:
Here is a code snippet from MainWindow.vala:
```
 directory_path = Environment.get_home_dir () + "/.local/share/applications";
        GLib.File file = GLib.File.new_for_path (directory_path);
        if (!file.query_exists ()) {
            alert (_("Error!\nPath %s does not exists!\nThe program will not be able to perform its functions.").printf(directory_path));
            button_create.sensitive = false;
            button_show_all.sensitive = false;
        }
```
Creating a file in a directory:
```
 string path = directory_path + "/" + entry_name.text + ".desktop";
        try {
            FileUtils.set_contents (path, desktop_file);
        } catch (Error e) {
            stderr.printf (_("Error: %s") + "\n", e.message);
        }

        GLib.File file = GLib.File.new_for_path (path);
        if (file.query_exists ()) {
            alert (_("App has been successfully created!"));
        } else {
            alert (_("Error! Could not create app"));
        }
```

Reply:
xdg-desktop-portal ships with a desktop launcher portal these days; I guess that's what this app should use.

Reply:
I replaced it with this:
` "--filesystem=xdg-data/applications:create"`

Reply:
It doesn't change my point – using a portal would allow dropping it from the app altogether.

Reply:
I can use this path to place icons:
~/.local/share/flatpak/exports/share/applications?

Reply:
For some reason this code is not working. Displays a message about a non-existent path:
```
directory_path = Environment.get_home_dir() + "/.local/share/flatpak/exports/share/applications";
        GLib.File file = GLib.File.new_for_path (directory_path);
        if (!file.query_exists ()) {
            alert (_("Error!\nPath %s does not exists!\nThe program will not be able to perform its functions.").printf(directory_path));
            button_create.sensitive = false;
            button_show_all.sensitive = false;
        }
```
The directory in this path exists. Checked. What's wrong?

Reply:
It seems to me that it is better to use a folder .local/share/applications, since the above folder is intended for flatpak.

Reply:
I've never worked with portals. Please tell me using the example of the above code how to do this. Or maybe everything should be left as it was?